tls/s2n_early_data_io.c

```diff
@@ -180,7 +180,8 @@ S2N_RESULT s2n_send_early_data_impl(struct s2n_connection *conn, const uint8_t *
      * We only care about the result of this call if it fails. */
     uint32_t early_data_to_send = 0;
     RESULT_GUARD_POSIX(s2n_connection_get_remaining_early_data_size(conn, &early_data_to_send));
-    early_data_to_send = MIN(data_len, early_data_to_send);
+    RESULT_ENSURE(data_len >= 0, S2N_ERR_SAFETY);
+    early_data_to_send = MIN((uint32_t) data_len, early_data_to_send);
```

Comment:
Is uint32_t large enough to always fit ssize_t? I think this should maybe be uint64_t.